lib/screens/login_screen.dart

```diff
@@ -2,13 +2,45 @@ import 'package:flutter/gestures.dart';
 import 'package:flutter/material.dart';
 import 'package:font_awesome_flutter/font_awesome_flutter.dart';
 import 'package:scdao_mobile/constants/color_constants.dart';
+import 'package:http/http.dart' as http;
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'dart:convert';
+import 'dart:io';
 
-class LoginScreen extends StatefulWidget {
+class LoginScreen extends ConsumerStatefulWidget {
   @override
   _LoginScreenState createState() => _LoginScreenState();
 }
 
-class _LoginScreenState extends State<LoginScreen> {
+final loginRepositoryProvider =
+    Provider<LoginRepository>((ref) => LoginRepository());
+
+class LoginRepository {
```

Comment:
Repository classes should be in their own file. We want to keep business logic(state manipulation,api calls etc) separate from the frontend code("presenter)